src/plugins/intel_gna/frontend/layer_quantizer.hpp

```diff
@@ -105,615 +97,374 @@ struct FakeQuantizeParams {
-        IE_THROW(NotAllocated)
-            << "[GNAPlugin] in function " << __PRETTY_FUNCTION__<< ": "
-            << "cannot copy weights for layer :"<< conv->name << " of size" << intWeights->byteSize();
-    }
-
-    auto getBiasSizeForLayer = [](InferenceEngine::WeightableLayer *wl) -> size_t {
-        if (wl->_biases) {
-            return wl->_biases->size();
-        }
-        // calculating biases len using outdata dims: biases number should be equal to output channels number
-        return InferenceEngine::GetDataDimSize(wl->outData.front(), InferenceEngine::DataDimName::C);
-    };
-
-    using BiasesPrecision = typename QuantDesc::BiasesPrecision;
-    auto biasMaker = [&] () {
+template <class BiasesType>
+void QuantizeBiasesPrep(InferenceEngine::WeightableLayer* wl, QuantizationData& common_data) {
+    const auto bias_maker = [&]() {
         InferenceEngine::Blob::Ptr zero;
-        if (!conv->_biases && !shouldAlwaysAllocate<BiasesPrecision>()) {
+        if (!wl->_biases && !ShouldAlwaysAllocate<BiasesType>()) {
             return zero;
         }
-        auto bias = make_custom_blob<BiasesPrecision>(InferenceEngine::C, InferenceEngine::SizeVector({
-                                                                                                          getBiasSizeForLayer(conv)
-                                                                                                      }));
+
+        InferenceEngine::Blob::Ptr bias = InferenceEngine::make_shared_blob<BiasesType>(
+            InferenceEngine::TensorDesc(InferenceEngine::Precision::fromType<BiasesType>(),
+                                        InferenceEngine::SizeVector({GetBiasSizeForLayer(wl)}),
+                                        InferenceEngine::C));
+
         bias->allocate();
+
         if (bias->buffer() == nullptr) {
-            IE_THROW(NotAllocated)
-                << "[GNAPlugin] in function " << __PRETTY_FUNCTION__<< ": "
-                << "cannot copy bias for layer :"<< conv->name <<"of size" << bias->byteSize();
+            IE_THROW(NotAllocated) << "[GNAPlugin] in function " << __PRETTY_FUNCTION__ << ": "
+                                   << "cannot copy bias for layer :" << wl->name << "of size" << bias->byteSize();
         }
+
         memset(bias->buffer(), 0, bias->byteSize());
 
         return bias;
     };
-    auto intBiases = biasMaker();
 
+    const auto int_biases = bias_maker();
+    common_data.scale_factor = InferenceEngine::getInjectedData<QuantizedLayerParams>(*wl)->_dst_quant.GetScale();
+
+    QuantizeBiases<BiasesType>(common_data,
+                               wl->_biases ? wl->_biases->buffer().as<float*>() : nullptr,
+                               int_biases ? int_biases->buffer() : static_cast<BiasesType*>(nullptr));
+
+    wl->_biases = int_biases;
+}
+
+static std::map<InferenceEngine::Precision, void (*)(InferenceEngine::WeightableLayer*, QuantizationData&)> GetWeightsPreicisionsMap() {
```

Comment:
```suggestion
const std::map<InferenceEngine::Precision, void (*)(InferenceEngine::WeightableLayer*, QuantizationData&)>& GetWeightsPreicisionsMap() {
    static std::map<InferenceEngine::Precision, void (*)(InferenceEngine::WeightableLayer*, QuantizationData&)> weight_precision_map = {{InferenceEngine::Precision::I8, QuantizeWeightsPrep<int8_t>},
            {InferenceEngine::Precision::I16, QuantizeWeightsPrep<int16_t>}};
    return weight_precision_map;
}
```
// currently each invocation creates new map

Reply:
Static needs to be left for now, so this suggestion won't work directly. Also [] operator needs to be replaced. But I'll change to const.